### PR TITLE
Adding helm chart for m3db-operator

### PR DESCRIPTION
**Pull request by nerd0**
Provides a simple helm chart for the `m3db-operator`

### CODE REVIEW

File: helm/m3db-operator/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+name: m3db-operator
+version: 0.0.1
+# TODO(PS) - helm has issues with GKE's SemVer
+# Error: Chart requires kubernetesVersion: >=1.10.6 which is incompatible with Kubernetes v1.10.7-gke.2
+#
+#kubeVersion: ">=1.10.7"
+description: m3db-operator provides an automated control plane for managing m3 clusters
```
Comment (schallert) [inline helm/m3db-operator/Chart.yaml]: Can we just say "managing m3db clusters" for now and once we manage full instances of the M3 stack we can change it accordingly?

File: helm/m3db-operator/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+### Helm Charts for M3 on Kubernetes
```
Comment (schallert) [inline helm/m3db-operator/README.md]: Same comment here re: the fact that we only manage the dbnodes themselves so far and not query service / aggregator etc.

Comment (nerd0) [reply helm/m3db-operator/README.md]: ditto

File: helm/m3db-operator/Chart.yaml (hunk 1 above)
Comment (nerd0) [reply helm/m3db-operator/Chart.yaml]: sure fair given the progress

File: helm/m3db-operator/NOTES.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+| | | | '_ \ / _ \ '__/ _` | __/ _ \| '__| | '_ \ / _` / __|
+| |_| | |_) |  __/ | | (_| | || (_) | |    | | | | (_| \__ \
+ \___/| .__/ \___|_|  \__,_|\__\___/|_|    |_| |_|\__,_|___/
+      |_|                                                   
+ _                       _           _        _ _          _ 
+| |__   ___  ___ _ __   (_)_ __  ___| |_ __ _| | | ___  __| |
+| '_ \ / _ \/ _ \ '_ \  | | '_ \/ __| __/ _` | | |/ _ \/ _` |
+| |_) |  __/  __/ | | | | | | | \__ \ || (_| | | |  __/ (_| |
+|_.__/ \___|\___|_| |_| |_|_| |_|___/\__\__,_|_|_|\___|\__,_|
+
```
Comment (CK-Ward) [inline helm/m3db-operator/NOTES.txt]: sick

**Comment by CK-Ward** [comment]
Closes #48

**Pull request merged**